### PR TITLE
Update taint mode introduction, minor edits in taint labels

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Comment by adamkvitek** [comment]
@IagoAbal many interesting points were made in https://github.com/returntocorp/semgrep-docs/pull/698#discussion_r969471876 - I decided to create this PR to clarify some of them. Let me know if this PR is fine.

File: docs/experiments/taint-labels.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ To include taint labels into a taint mode rule, follow these steps:
             requires: INPUT
             label: EVIL
     ```
-    The `requires` key accepts a Python Boolean expression over labels. For example: `requires: LABEL1 and not LABEL2`.
+    Combine labels using `requires` key. To combine labels, use Python Boolean operators. For example: `requires: LABEL1 and not LABEL2`.
```
Comment (IagoAbal) [inline docs/experiments/taint-labels.md:30]:
```suggestion
    Combine labels using the `requires` key. To combine labels, use Python Boolean operators. For example: `requires: LABEL1 and not LABEL2`.
```

File: docs/experiments/taint-labels.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ To include taint labels into a taint mode rule, follow these steps:
 
 :::info
 - Semgrep accepts valid Python identifiers as labels.
-- Restrict a source to a subset of labels using the `requires` key. You can also use Python Boolean expressions instead. For example: `requires: LABEL1 and not LABEL2`.
+- Restrict a source to a subset of labels using the `requires` key. You can combine more labels in `requires` key using Python Boolean operators. For example: `requires: LABEL1 and not LABEL2`.
```
Comment (IagoAbal) [inline docs/experiments/taint-labels.md:41]:
```suggestion
- Restrict a source to a subset of labels using the `requires` key. You can combine more labels in the `requires` key using Python Boolean operators. For example: `requires: LABEL1 and not LABEL2`.
```

**Review by IagoAbal — APPROVED** [review]
Looks good!

**Pull request merged**